pylint/checkers/variables.py

```diff
@@ -1543,7 +1543,10 @@ def _check_consumer(
                     )
 
         elif self._is_only_type_assignment(node, defstmt):
-            self.add_message("undefined-variable", args=node.name, node=node)
+            if node.scope().locals.get(node.name):
+                self.add_message("used-before-assignment", args=node.name, node=node)
+            else:
+                self.add_message("undefined-variable", args=node.name, node=node)
```

Comment:
Maybe we could start adding the confidence if it's easy enough to determine, what do you think ?